include/aspect/utilities.h

```diff
@@ -1060,6 +1066,41 @@ namespace aspect
      * entry in the list must match one of the allowed operations.
      */
     std::vector<Operator> create_model_operator_list(const std::vector<std::string> &operator_names);
+
+    /**
+    * A class that represents a point in a chosen coordinate system.
+    */
+    template <int dim>
+    class GeneralizedPoint
+    {
+      public:
+        GeneralizedPoint(Point<dim> position, const GeometryModel::Interface<dim> *geometry_model);
```

Comment:
pass position as ``const &``

Reply:
and break the arguments onto separate lines.

Reply:
I think the constructor could also use a doc string :-)

Reply:
By the way, I tend to think that it would be nicer from a conceptual point of view if the class did not know about geometry models. You are here introducing a circular dependency, because this class needs to know about `GeometryModel::Interface`, while that class also needs to know about `GeneralizedPoint`.

You can avoid this if the class only takes as argument a `CoordinateSystem` argument as second argument. It would *typically* be filled by asking the geometry model what the appropriate coordinate system is, but you'd break the circular dependency.

Reply:
I agree that it would be nicer to use the enum CoordinateSystem as an argument and I see the circular dependency problem (that is why I need to specifically declare the type of the enum), but I am not sure if I understand what you propose as a solution.

The problem I face with making the CoordinateSystem as an input is twofold. The first problem is that is that there is no direct linkage between the enum CoordinateSystem and the geometrymodels. Multiple geometrymodels belong to one CoordinateSystem. 

And secondly, related to that, access to the geometry model will be needed anyway, and from just a enum CoordinateSystem I can't start a new geometry model to do the transformation, because for example the ellipsoidal geometrymodel needs to fill it's manifold with parameters like eccentricity etc. (It would also seem wasteful to me to just start a new geometry model to transform points.).